ui-tests/src/test/java/org/openmrs/reference/page/HeaderPage.java

```diff
@@ -17,13 +19,13 @@ public void clickOnHomeIcon() throws InterruptedException {
         clickOn(By.className(HOME_ICON));
     }
 
-
     // TODO This is unused, do we really need it?
     public void clickOnHomeLink() {
         clickOn(By.linkText(HOME_LINK_TEXT));
     }
 
     public void logOut() throws InterruptedException {
+        waiter.until(ExpectedConditions.visibilityOfElementLocated(LOGOUT));
```

Comment:
wouldn't this be part of login test?

Reply:
This is needed here as well  for all pages that consumes HeaderPage, so the first commit to merge having this change will have to resolve merge conflict